boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk

```diff
@@ -2,9 +2,16 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_NRF54H20DK
-	select SOC_NRF54H20_CPUAPP if BOARD_NRF54H20DK_NRF54H20_CPUAPP
-	select SOC_NRF54H20_CPURAD if BOARD_NRF54H20DK_NRF54H20_CPURAD
+	select SOC_NRF54H20_CPUAPP if (BOARD_NRF54H20DK_NRF54H20_CPUAPP || \
+				       BOARD_NRF54H20DK_NRF54H20_CPUAPP_IRON)
 	select SOC_NRF54H20_CPUPPR if (BOARD_NRF54H20DK_NRF54H20_CPUPPR || \
 				       BOARD_NRF54H20DK_NRF54H20_CPUPPR_XIP)
 	select SOC_NRF54H20_CPUFLPR if (BOARD_NRF54H20DK_NRF54H20_CPUFLPR || \
 				        BOARD_NRF54H20DK_NRF54H20_CPUFLPR_XIP)
+
+config NRF_IRON
+	bool
+	default y if BOARD_NRF54H20DK_NRF54H20_CPUAPP_IRON
+	help
+	  Temporary board variant used to build local domain firmware compatible with
+	  Nordic IRONside SE.
```

Comment:
why is this here?

Reply:
I use this to indicate that the iron variant is in use since it affects parts of the build system. Is there another way I can identify that this variant is used in cmake?

Reply:
discussed offline: move to https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nordic/nrf54h/Kconfig.soc

As 

```
config SOC_NRF54H20_IRON
```